stan/math/prim/scal/meta/broadcast_array.hpp

```diff
@@ -14,18 +14,18 @@ namespace stan {
       public:
         explicit broadcast_array(T& prim): prim_(prim) {}
 
-        T& operator[] (int /*i*/) {
+        T& operator[](int /*i*/) {
           return prim_;
         }
       };
 
-      template <typename T>
+      template <typename T, typename S>
       class empty_broadcast_array {
       public:
         empty_broadcast_array() {}
-        T& operator[] (int /*i*/) {
-          throw std::logic_error("Don't do this");
-        }
+        // We provide a stub method for the empty_broadcast_array which should
```

Comment:
This should just be regular doxygen doc and not written in first person like this.  As in:  `Not implemented so cannot be called`.